pkg/dockerregistry/server/client/client.go

```diff
@@ -0,0 +1,162 @@
+type RegistryClient interface {
+	// Client returns the authenticated client to use with the server.
+	Client() (Interface, error)
+	// ClientFromToken returns a client based on the user bearer token.
+	ClientFromToken(token string) (Interface, error)
+}
+
+// Interface contains client methods that registry use to communicate with
+// Origin or Kubernetes API.
+type Interface interface {
+	ImageSignaturesInterfacer
+	ImagesInterfacer
+	ImageStreamImagesNamespacer
+	ImageStreamMappingsNamespacer
+	ImageStreamSecretsNamespacer
+	ImageStreamsNamespacer
+	ImageStreamTagsNamespacer
+	LimitRangesGetter
+	LocalSubjectAccessReviewsNamespacer
+	SubjectAccessReviews
+	UsersInterfacer
+}
+
+type apiClient struct {
+	oc    client.Interface
+	kube  kcoreclient.CoreInterface
+	auth  authclientv1.AuthorizationV1Interface
+	image imageclientv1.ImageV1Interface
+	user  userclientv1.UserV1Interface
+}
+
+func newAPIClient(
+	c client.Interface,
+	kc kcoreclient.CoreInterface,
+	authClient authclientv1.AuthorizationV1Interface,
+	imageClient imageclientv1.ImageV1Interface,
+	userClient userclientv1.UserV1Interface,
+) Interface {
+	return &apiClient{
+		oc:    c,
+		kube:  kc,
+		auth:  authClient,
+		image: imageClient,
+		user:  userClient,
+	}
+}
+
+func (c *apiClient) Users() UserInterface {
+	return c.user.Users()
+}
+
+func (c *apiClient) Images() ImageInterface {
+	return c.image.Images()
+}
+
+func (c *apiClient) ImageSignatures() ImageSignatureInterface {
+	return c.image.ImageSignatures()
+}
+
+func (c *apiClient) ImageStreams(namespace string) ImageStreamInterface {
+	return c.image.ImageStreams(namespace)
+}
+
+func (c *apiClient) ImageStreamImages(namespace string) ImageStreamImageInterface {
+	return c.image.ImageStreamImages(namespace)
+}
+
+func (c *apiClient) ImageStreamMappings(namespace string) ImageStreamMappingInterface {
+	return c.image.ImageStreamMappings(namespace)
+}
+
+func (c *apiClient) ImageStreamTags(namespace string) ImageStreamTagInterface {
+	return c.image.ImageStreamTags(namespace)
+}
+
+func (c *apiClient) ImageStreamSecrets(namespace string) ImageStreamSecretInterface {
+	return c.oc.ImageStreamSecrets(namespace)
+}
+
+func (c *apiClient) LimitRanges(namespace string) LimitRangeInterface {
+	return c.kube.LimitRanges(namespace)
+}
+
+func (c *apiClient) LocalSubjectAccessReviews(namespace string) LocalSubjectAccessReviewInterface {
+	return c.auth.LocalSubjectAccessReviews(namespace)
+}
+
+func (c *apiClient) SubjectAccessReviews() SubjectAccessReviewInterface {
+	return c.auth.SelfSubjectAccessReviews()
+}
+
+type registryClient struct {
+	config *clientcmd.Config
+}
+
+func NewRegistryClient(config *clientcmd.Config) RegistryClient {
+	return &registryClient{config: config}
+}
+
+// Client returns the authenticated client to use with the server.
+func (c *registryClient) Client() (Interface, error) {
+	oc, kc, err := c.config.Clients()
+	if err != nil {
+		return nil, err
+	}
+
+	return newAPIClient(
+		oc,
+		kc.Core(),
```

Comment:
why aren't these constructed like the others?

Reply:
agree, we don't have force us to use clientmd